[DOCS] DOC-414: Remove guide for use of outdated docker images

Rachel-Reverie opened the pull request:
Changes proposed in this pull request:
- Remove out dated guide (directing to use of out dated docker images)
- Redirect guide URL to the setup overview page-

### Definition of Done
- [X] I have made corresponding changes to the documentation

Rachel-Reverie committed:
DOC-414
- Removes out dated guide (directing to use of out dated docker images)
- Redirects guide URL to the setup overview page

ghost commented:
<details>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=6718&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

  - Make big changes easier — review code in small groups of related files
  - Know where to start — see the whole change at a glance
  - Take a code tour — explore the change with an interactive tour
  - Make comments and review — all fully sync’ed with github

    [Try it now!](https://app.codesee.io/r/reviews?pr=6718&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)
</details>

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/6718/5c5a65dc/9a7872b77e03006c319ac2e8d5109cba325c0487.svg)](https://app.codesee.io/r/reviews?pr=6718&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

Rachel-Reverie committed:
DOC-414
- Updates the ToC and setup index pages, removes "miscellaneous" section from each under setup

Rachel-Reverie committed:
DOC-414
- Removes import of miscellaneous index section

Rachel-Reverie committed:
DOC-414
- fix typo in redirects

Rachel-Reverie committed:
Merge branch 'develop' into d/docs/doc-414/remove_outdated_how_to_use_docker_images_guide

donaldheppner committed:
Merge branch 'develop' into d/docs/doc-414/remove_outdated_how_to_use_docker_images_guide
